dm/syncer/syncer.go

```diff
@@ -1012,23 +1012,24 @@ func (s *Syncer) handleJob(job *job) (err error) {
 
 	if waitXIDStatus(s.waitXIDJob.Load()) == waitComplete && job.tp != flush {
```

Comment:
you mean in this line return nil-err and outer logic call s.checkShouldFlush() ? and this cause checkpoint have wrong pos ?

Reply:
yes.